client_src/js/services/octoprint-client.service.js

```diff
@@ -0,0 +1,927 @@
+
+        const post = await OctoPrintClient.post(printer, "plugin/pluginmanager", postData);
+        alert.close();
+        if (post.status === 409) {
+          UI.createAlert(
+            "error",
+            "Plugin not installed... Printer could be active...",
+            4000,
+            "Clicked"
+          );
+        } else if (post.status === 400) {
+          UI.createAlert("error", "Malformed request... please log an issue...", 4000, "Clicked");
+        } else if (post.status === 200) {
+          let response = await post.json();
+          if (response.needs_restart || response.needs_refresh) {
+            UI.createAlert(
+              "success",
+              `${printer.printerName}: ${pluginList[r]} - Has successfully been installed... OctoPrint restart is required!`,
+              4000,
+              "Clicked"
+            );
+          } else {
+            UI.createAlert(
+              "success",
+              `${printer.printerName}: ${pluginList[r]} - Has successfully been installed... No further action requested...`,
+              4000,
+              "Clicked"
+            );
+          }
+        }
+      }
+    } else {
+      UI.createAlert(
+        "danger",
+        `${printer.printerName}: Is active skipping the plugin installation command...`
+      );
+    }
+  }
+  static async updateOctoPrintPlugins(pluginList, printer) {
+    const data = {
+      targets: pluginList,
+      force: true
+    };
+    let updateRequest = await OctoPrintClient.postNOAPI(
+      printer,
+      "plugin/softwareupdate/update",
+      data
+    );
+    if (updateRequest.status === 200) {
+      UI.createAlert(
+        "success",
+        `${printer.printerName}: Successfully updated! your instance will restart now.`,
+        3000,
+        "Clicked"
+      );
+      let post = await OctoPrintClient.systemNoConfirm(printer, "restart");
+      if (typeof post !== "undefined") {
+        if (post.status === 204) {
+          UI.createAlert(
+            "success",
+            `Successfully made restart attempt to ${printer.printerName}... You may need to Re-Sync!`,
+            3000,
+            "Clicked"
+          );
+        } else {
+          UI.createAlert(
+            "error",
+            `There was an issue sending restart to ${printer.printerName} are you sure it's online?`,
+            3000,
+            "Clicked"
+          );
+        }
+      } else {
+        UI.createAlert(
+          "error",
+          `No response from ${printer.printerName}, is it online???`,
+          3000,
+          "Clicked"
+        );
+      }
+    } else {
+      UI.createAlert(
+        "error",
+        `${printer.printerName}: Failed to update, manual intervention required!`,
+        3000,
+        "Clicked"
+      );
+    }
+  }
+
+  static async updateOctoPrintClient(printer) {
+    const data = {
+      targets: ["octoprint"],
+      force: true
+    };
+    let updateRequest = await OctoPrintClient.postNOAPI(
```

Comment:
`postNOAPI` is not defined. Probably commented out on the bottom.

Reply:
Yep, not checked off yet... I'm aware.